aea/cli/upgrade.py

```diff
@@ -139,6 +137,20 @@ def _update_agent_config(ctx: Context):
     ctx.dump_agent_config()
 
 
+def update_aea_version_in_nonvendor_packages(ctx: Context):
```

Comment:
I'd remove ctx and only have `working_dir` as arg, makes it reusable without ctx.